src/libsync/graphapi/spacesmanager.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
+#include "spacesmanager.h"
+
+#include "drives.h"
+
+#include "account.h"
+#include "creds/abstractcredentials.h"
+
+#include <QTimer>
+
+#include <chrono>
+
+using namespace std::chrono_literals;
+
+using namespace OCC;
+using namespace GraphApi;
+
+namespace {
+constexpr auto refreshTimeoutC = 30s;
+}
+
+SpacesManager::SpacesManager(Account *parent)
+    : QObject(parent)
+    , _account(parent)
+    , _refreshTimer(new QTimer(this))
+{
+    _refreshTimer->setInterval(refreshTimeoutC);
+    _refreshTimer->setSingleShot(true);
```

Comment:
Maybe you should document that you use the single shot mode and (re)start the timer manually to make sure the HTTP request in the handler has finished.

Reply:
This line was irritating me, so I looked on and found you would (re)start the timer manually below. That worked only because I've seen similar code in the codebase, though. I'd put such a comment right above this line.